src/m365/entra/commands/pim/pim-role-assignment-list.ts

```diff
@@ -0,0 +1,168 @@
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupName: typeof args.options.groupName !== 'undefined',
+        startDateTime: typeof args.options.startDateTime !== 'undefined',
+        includePrincipalDetails: !!args.options.includePrincipalDetails
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "--userId [userId]"
+      },
+      {
+        option: "--userName [userName]"
+      },
+      {
+        option: "--groupId [groupId]"
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "-s, --startDateTime [startDateTime]"
+      },
+      {
+        option: "--includePrincipalDetails [includePrincipalDetails]"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.startDateTime && !validation.isValidISODateTime(args.options.startDateTime)) {
+          return `${args.options.startDateTime} is not a valid ISO 8601 date time string`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({
+      options: ['userId', 'userName', 'groupId', 'groupName'],
+      runsWhen: (args) => args.options.userId || args.options.userName || args.options.groupId || args.options.groupName
+    });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const queryParameters: string[] = [];
+    const filters: string[] = [];
+
+    try {     
+      const principalId = await this.getPrincipalId(logger, args.options);
+            
+      if (principalId) {
+        filters.push(`principalId eq '${principalId}'`);
+      }
+
+      if (args.options.startDateTime) {
+        filters.push(`startDateTime ge ${args.options.startDateTime}`);
+      }
+
+      if (filters.length > 0) {
+        queryParameters.push(`$filter=${filters.join(' and ')}`);
+      }
+
+      if (args.options.includePrincipalDetails) {
+        queryParameters.push('$expand=principal');
+      }
+
+      const queryString = queryParameters.length > 0
+        ? `?${queryParameters.join('&')}`
+        : '';
+
+      const url = `${this.resource}/v1.0/roleManagement/directory/roleAssignmentScheduleInstances${queryString}`;
```

Comment:
Hi @MartinM85, I found one last thing... In the issue specs I'm mentioning that it's nice to know what the role definition name is out of hand. So we need to expand the roleDefinition. Could you add that still?

> By default, we need to expand/request the displayName of the role definition when calling the endpoint, using ?$expand=roleDefinition($select=displayName)

Reply:
You don't need to flatten the response object though. Just leave it as an object.